### ISSUE TITLE
Feature request: Be able to set fixed cost depending on time

### ISSUE THREAD

**Issue by kmengvik** [gruijter/com.gruijter.powerhour#27]
For the power meter devcie. Is it possible to add a then card to reset the meter? ie when power becomes greater than, then reset meter value.

**Comment by gruijter** [comment]
Can you elaborate on the exact use case? Im not following what you want to achieve.

**Comment by kmengvik** [comment]
To measure energy consumption for a device from start to stop. When power (from the device) become greater than 10W, then reset meter value. 

When cycle is finished, the meter value will output to simple log in a different flow.

**Comment by gruijter** [comment]
I assume you want to reset the kWh for a device that uses Watts as source?
Does this mean you are using PBTH to create a kWh meter from a watt-only device?

Still I would like to understand your use case better. What is the device you are talking about, and why would it cycle when the power reaches a certain level? And if it does cycle, why would you want the total kWh to reset? My car charger cycles, but at the end of the year I still want to know how much kWh in total was used...

**Comment by kmengvik** [comment]
Thank you for following this up closely.

Yes, the device has W source. The idle effect is 7 W.

[image: image.png]

As far as I can tell, I can set up as many independent virtual PBTH devices
as I want. Therefore I can set up both an total acumulative meter, and one
for each run.

In this particular case, I have a smart plug that measure momentarily
effect on my thumble dryer. I want to reset kwh value at each start with a
then card. In a different flow, I will when the power drops below 10W (idle
again) output the current meter value to simple log. This way I can
document energy usage of different programs used by the dryer.

Kai

fre. 24. jun. 2022 kl. 08:23 skrev gruijter ***@***.***>:

**Comment by kmengvik** [comment]
I also have a request for the tariff specification.

In our country it is a bit complicated. In addition to the hourly spot
price, we also have a grid price. This is also different night and day. So
I need to be able to input this information to the tarif spec:
[image: image.png]

[image: image.png]

But it needs to automatically change according to night and day.

fre. 24. jun. 2022 kl. 08:23 skrev gruijter ***@***.***>:

**Comment by gruijter** [comment]
Mmh ok. So I gave your 1st request some deep thoughts. But I decided against it. Your use case is very on the edge. The core function of PBTH is to summarize usage/cost per hour,day,month, year. Resetting the kWh meter completely messes up this function. Besides that, what you want to achieve you can already do now with logic cards and a variable that stores the last kWh value.

Your second request is much more in line with what I want to do with PBTH. So I will put this on the wish-list.

Thx for your input!

**Comment by kmengvik** [comment]
Ok, so you say I can achieve what I want in the first request with logic? I
have no skills with variables yet. Do you have a handy example on how to
go about this particular example?

Kai

fre. 24. jun. 2022 kl. 10:10 skrev gruijter ***@***.***>:

**Comment by kmengvik** [comment]
Think I managed to go about it :)

[image: image.png]

fre. 24. jun. 2022 kl. 10:10 skrev gruijter ***@***.***>:

**Comment by kmengvik** [comment]
In the day ahead device, would it be possible to fetch the exchange rate automatically. It is not likely I would enter the days rate manually every day :)

**Comment by gruijter** [comment]
Tariff change and markup change can now be done from a flow. 

PLease test v 4.3.6 and let me know if it works: https://homey.app/a/com.gruijter.powerhour/test

PS: No, I will not fetch exchange rates automatically from PBTH. But you can now set it via a flow, so I'm sure you will find some clever way of doing it :)

**Comment by kmengvik** [comment]
Thanks.. Trying hard to decode what you are telling me here.. 

Line one: "Tariff change and markup change can now be done from a flow."

Not sure what you mean by refering to flow. As I interpret, I can enter addon fee to the tariff in the "day ahead device" settings, but no option to import from flow here as far as I can tell?

And, In my country, this addon is time of day differented, so I need to add on a fee for time between 22-06, and a different addon fee between 06-22.

The PS: If I find a flow card for exhcange rates in another app, are you saying i can output that flow into the exchange rate of the "Day ahead device"

Please be patient if I discribed all this poorly.

![markup](https://user-images.githubusercontent.com/16176305/176170624-372ae47e-d7b1-4738-995a-bedf3df34aab.JPG)
?

**Comment by gruijter** [comment]
See this example flow (although it is more reliable to make a second flow for 6am): https://homey.app/nl-nl/flow/1WbkG6/

It works similarly for exchange rate, and you can indeed use a 'tag' from an app that delivers the rate.

**Comment by kmengvik** [comment]
Great, thanks! Did not find any currency tag anywhere. But why do I need to bother with currency anyway? In the power device, I am allowed to import the tariff in local currency. Why not the same in the "day ahead" decice?

**Comment by kmengvik** [comment]
really?

tir. 28. jun. 2022 kl. 17:06 skrev gruijter ***@***.***>:

**Comment by gruijter** [comment]
I closed the issue since I thought you confirmed the correct working of the requested feature 'Be able to set fixed cost depending on time'. If it is not working correctly please reopen this issue and let me know what exactly is not working. If it does work correctly, I do hope you appreciate the work I put in based on your feature request!

If you have another issue or another feature request please open a new issue. If you need help on 'how to use' a feature, please go to the forum and post your help request there.

**Comment by kmengvik** [comment]
Yes, I'm sorry.. probably to many open issues :D Do we still have a
conversation on the currency issue? Can we not just import the price from a
flow like in the power device?

Kai

tir. 28. jun. 2022 kl. 17:47 skrev gruijter ***@***.***>:

**Comment by gruijter** [comment]
Discussing different topics under one issue/feature request is making managiement of all my active requests/issues extremely hard. E.g. I closed this issue because it is solved. And when it is closed it disappears from my list...

But to answer your question: no I wont add the ability to import pricing via a flow. The DAP driver needs 3 x 24 hours worth of pricing to work (yesterday, today, tomorrow).

**Comment by kmengvik** [comment]
Hey! I was using the 4.3.6, with day/night markup. Today it is gone, and
all my price settings are lost. Can it be rolled back?

tir. 28. jun. 2022 kl. 12:26 skrev gruijter ***@***.***>:

**Comment by gruijter** [comment]
What is gone? I only added features, not removed anything.

**Comment by gruijter** [comment]
And you did read my post about the changes to fixed cost in version 4.4.1? https://community.homey.app/t/app-pro-power-by-the-hour-insights-per-hour-day-month-and-year/20535/660

**Comment by kmengvik** [comment]
Thank you again. I am ever so grateful for your first class support. A true hero.
No, I did not notice the post before. It explains many things.
From the post: 
«During migration to 4.4.1 I already recalculated the markup for you.»
This did not pay off in my particular case. As we remember, at least I do, I have flows that set different markup day/night. Here the markup was nominated in Euro in the flows. So whatever was done in migration, was overwritten again by the flows. So my many flows for pricing different sessions were totally screwed. 
The update explains why I got a mysterious push yesterday. I have a flow that notifies me when the price 3+ hours goes sky high. When the currency was changed from Euro to local, the flow triggered, and sent me the push. I did not understand why then, but now I do.
While we are at this specific point, I would like to address an issue which is relevant at least for the Norwegian users, who have day/night prices. When we close up on the clock to the time when price changes, the price + hours will not show correct price beyond the price change time, until the time passes, and the markup is updated from the flow. Is there any way to work around this? 
Another minor issue, which can be confusing, is that on the flow card that sets markup, the € is hardcoded. Can this be set as a dropdown, where we can set our local currency?
